metaflow/plugins/aws/batch/batch_decorator.py

```diff
@@ -109,6 +109,10 @@ def myStep(self):
     run_time_limit = None
 
     def __init__(self, attributes=None, statically_defined=False):
+        username = util.get_username()
+        if not re.match('^[a-zA-Z0-9][a-zA-Z0-9_-]*$', username):
+            raise BatchException('The username trying to execute this batch job has one or more illegal characters. Make sure the first character in the username is alphanumeric and that it only contains dashes, underscores or alphanumeric characters. You can also set a specific username via the USER environment variable')
```

Comment:
This seems pretty verbose for the end-user to parse easily. 
Maybe we can throw a `BatchInvalidUserException` (similar to [this](https://github.com/Netflix/metaflow/blob/c2f70c73782549fdce6c418410ecd9b305199b28/metaflow/exception.py#L84)) with the headline - 
`Invalid user name for launching jobs on AWS Batch.` 
and the formatted message 
`User name (overridden via METAFLOW_USER environment variable) should obey the following constraints - `

    <tabbed>1. It should only contain dashes, underscores or alphanumeric characters
    <tabbed>2. The first character should be alphanumeric

Reply:
Or the message could be simply - `The user name (overridden via METAFLOW_USER environment variable) can only contain lowercase letters, numbers, underscores and dashes (and must start with a letter or number).`